src/middlewared/middlewared/plugins/etc.py

```diff
@@ -366,11 +346,14 @@ def opener(path, flags):
         if outfile_dirname != '/etc':
             os.makedirs(outfile_dirname, exist_ok=True)
 
-        with open(full_path, "w", opener=opener) as f:
-            perms_changed = self.set_etc_file_perms(f.fileno(), entry)
-            contents_changed = write_if_changed(f.fileno(), rendered)
+        payload = self.get_perms_and_ownership(entry)
+        try:
+            changes = write_if_changed(full_path, rendered, raise_error=True, **payload)
+        except UnexpectedFileChange as e:
+            self.logger.error(str(e))
```

Comment:
Do we not just want to do `exc_info=True` here?

Reply:
We should also probably catch all exceptions here actually.

Reply:
Yeah, that's a good idea, I got side-tracked by writing tests.